reference-service/src/main/java/com/transformuk/hee/tis/reference/service/api/RoleResource.java

```diff
@@ -398,4 +399,26 @@ public ResponseEntity<List<RoleDTO>> getRolesIn(@PathVariable String codes) {
     }
     return new ResponseEntity<>(resp, HttpStatus.OK);
   }
+
+  /**
+   * GET /roles/codes : get roles given to codes.
+   *
+   * @param codes the codes to search by, using a ',' separator
+   * @return the ResponseEntity with status 200 (OK) and with body the list of roleDTOs, or empty
+   *     list
+   */
+  @GetMapping("/roles/codes")
+  public ResponseEntity<List<RoleDTO>> getRolesByCodes(
+      @RequestParam(value = "codes", required = false, defaultValue = "") String codes) {
```

Comment:
1. If `required = true` and the param is not specified in the url, Http 500 will be sent by default. The default value could help avoid this 500 error.
2. I've checked Generick Upload and TCS, before calling `findRolesIn`(in reference-client), they check `StringUtils.isNotEmpty(codes)` .